warp/tests/test_options.py

```
@@ -127,6 +136,97 @@ def test_options_opt_level(test, device):
+    frame.f_globals["__spec__"] = spec
+
+    with patch("warp._src.context._resolve_module_from_frame", return_value=None):
+        name = _caller_module_name(frame)
+
+    test.assertEqual(name, "fake_runpy_module")
+
+    # Restore globals
+    frame.f_globals.update(original_globals)
+
+
+def test_caller_module_name_fallback_to_dunder_name(test, device):
```

Ruff (0.15.5)

[warning] Unused function argument: `device`

(ARG001)